### PR TITLE
[DUOS-2931][risk=no] Implement v4 Algorithm

### PR DESCRIPTION
**Pull request by rushtong**
### Addresses
https://broadworkbench.atlassian.net/browse/DUOS-2931

### Summary
* Add a new set of endpoints for retrieving, validating and matching on a new V4 matching algorithm
* Adds documentation for the new algorithm
* This is essentially a copy of the V3 algorithm except that it drops `commercialUse` and relies explicitly on `nonProfitUse`
* Includes a minor bug fix (in V3) to remove duplicate reasons from the list of match reasons.

The primary documentation change is:

Screenshot 2024-02-15 at 4 49 56 PM

----
Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Review by aarohinadkarni — APPROVED** [review]
👍

File: src/main/java/org/broadinstitute/dsde/consent/ontology/util/JsonSchemaUtil.java [hunk 1]
```diff
@@ -62,12 +84,32 @@ public List<String> validateDataUseV3Schema(String dataUseV3Instance) {
...
+   * Compares an instance of a data use object to the data use V4 schema
+   *
+   * @param dataUseV4Instance The string instance of a data use object
+   * @return List of validation messages
+   */
+  public List<String> validateDataUseV4Schema(String dataUseV4Instance) {
+    try {
+      ObjectMapper mapper = new ObjectMapper();
+      JsonNode jsonSubject = mapper.readTree(dataUseV4Instance);
+      JsonSchema schema = getDataUseV3Instance();
```
Comment (fboulnois) [inline src/main/java/org/broadinstitute/dsde/consent/ontology/util/JsonSchemaUtil.java]:
```suggestion
      JsonSchema schema = getDataUseV4Instance();
```

I think this needs to be V4

**Review by fboulnois — COMMENTED** [review]
See below:

File: src/main/java/org/broadinstitute/dsde/consent/ontology/util/JsonSchemaUtil.java (hunk 1 above)
Comment (rushtong) [reply src/main/java/org/broadinstitute/dsde/consent/ontology/util/JsonSchemaUtil.java]: Nice catch!

**Review by fboulnois — APPROVED** [review]
looks good 👍

**Pull request merged**